src/OneExplorer/OneExplorer.ts

```diff
@@ -163,6 +163,45 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
     this._onDidChangeTreeData.fire();
   }
 
+  // TODO: Add move()
+
+  /**
+   * Rename a file of all types of nodes (baseModel, derivedModel, config) excepts for directory.
+   * It only alters the file name, not the path.
+   */
+  rename(oneNode: OneNode): void {
+    // TODO: prohibit special characters for security ('..', '*', etc)
+    let warningMessage;
+    if (oneNode.node.type === NodeType.baseModel) {
+      // TODO automatically change the corresponding files
+      warningMessage = 'WARNING: You may need to change input paths in the following cfg files:\n';
+      for (const conf of oneNode.node.childNodes) {
+        // NOTE A newline character is not supported in input box.
+        // vscode-extension doesn't plan to support multiple lines inside in input box or in
+        // notification box.
+        warningMessage += `${conf.name} `;
+      }
+    } else {
+      warningMessage = 'WARNING: Renaming may result in some unexpected changes on the tree view.';
+    }
+
+    vscode.window
+        .showInputBox({
+          title: 'Enter a file name:',
+          placeHolder: `${path.basename(oneNode.node.uri.fsPath)}`,
+          prompt: warningMessage
+        })
+        .then(newname => {
+          if (newname) {
+            const dirpath = path.dirname(oneNode.node.uri.fsPath);
+            const newpath = `${dirpath}/${newname}`;
+            vscode.workspace.fs.rename(oneNode.node.uri, vscode.Uri.file(newpath));
```

Comment:
How about giving proper extension name if it's not given by user? I cannot imagine the danger if we apply following change. 

```suggestion
          if (newname) {
            const dirpath = path.dirname(oneNode.node.uri.fsPath);
            const newpath = `${dirpath}/${newname}`;
            const ext = path.extname(oneNode.node.uri.fsPath);
            if (path.extname(newname) != ext) {
              newname = `${newname}.${ext}`;
            }
            vscode.workspace.fs.rename(oneNode.node.uri, vscode.Uri.file(newpath));
```

Reply:
IMO, it's a little bit too much... Maybe it will bring about much confusion to user. Well, how about just putting an error message instead?

Reply:
> IMO, it's a little bit too much... Maybe it will bring about much confusion to user.

I agree with you. Since there is a warning message at the bottom of input box, user would pay attention with his/her file naming. IMHO Defining the situation when the error message is popping up will be pretty difficult. 
Then, never mind about the above suggestion!

Reply:
@yunjayh I think we can use 'validateInput' API to check if user has given a right ext. I will apply that on the next pr. thanks for your opinion.